test/unit/dbUtils.test.js

```diff
@@ -0,0 +1,94 @@
+const { filterInvalid, buildSQLQuery } = require('../../db/utils');
+
+describe('Database query-building utilities', () => {
+  test('filterInvalid should filter input to remove invalid key-val pairs (without mutating)', () => {
+    // filterValid will only ever take an object with 0-all of the following keys:
+    // review_type, purchase_type, from, to, exclude, play_min, play_max, display
+    expect(filterInvalid({})).toStrictEqual({});
+    expect(filterInvalid({ invalidKey: 'invalidVal' })).toStrictEqual({});
+
+    let validValues = {
+      review_type: ['all', 'positive', 'negative'],
+      purchase_type: ['all', 'steam', 'other'],
+      display: ['summary', 'helpful', 'recent', 'funny'],
+      exclude: ['true', 'false']
+    };
+    validValues.review_type.forEach(validReview => {
+      validValues.purchase_type.forEach(validPurchase => {
+        validValues.display.forEach(validDisplay => {
+          validValues.exclude.forEach(validExclude => {
+            let validObj = {
+              review_type: validReview,
+              purchase_type: validPurchase,
+              display: validDisplay,
+              exclude: validExclude
+            };
+            expect(filterInvalid(validObj)).toStrictEqual(validObj);
+          });
+        });
+      });
+    });
+
+    // filterInvalid doesn't handle range checks, neither does buildSQLQuery.
```

Comment:
This reminds me that we may need to discuss URL parameters for implementing these queries.